core/capabilities/remote/trigger_publisher.go

```diff
@@ -54,12 +54,16 @@ func NewTriggerPublisher(config *capabilities.RemoteTriggerConfig, underlying co
 		config = &capabilities.RemoteTriggerConfig{}
 	}
 	config.ApplyDefaults()
+	workflowDONInfos := make(map[uint32]donInfo)
+	for id, don := range workflowDONs {
+		workflowDONInfos[id] = NewDONInfo(don)
+	}
 	return &triggerPublisher{
 		config:        config,
 		underlying:    underlying,
 		capInfo:       capInfo,
-		capDonInfo:    capDonInfo,
-		workflowDONs:  workflowDONs,
+		capDonInfo:    NewDONInfo(capDonInfo),
```

Comment:
Is this done just so you can create `.Members` map for easier checking below: `!callerDon.Members[sender]`?

Reply:
Yes exactly, I want to have a map for easier lookups (see comment above)

Reply:
Contains is O(n) on a list - I wanted to avoid that.